test_exceed_rate_limit is flaky

https://github.com/chatmail/relay/commit/ab3492d9a14399807514cdedb580a5f262fa8785 is merged,
but the test failed on `main`:
```
 ____________________________ test_exceed_rate_limit ____________________________
[gw0] linux -- Python 3.12.3 /home/runner/work/relay/relay/venv/bin/python3

cmsetup = <cmdeploy.tests.plugin.CMSetup object at 0x7f819c462480>
gencreds = <function gencreds.<locals>.<lambda> at 0x7f819c27a980>
maildata = <function maildata.<locals>.maildata at 0x7f819c27ade0>
chatmail_config = <chatmaild.config.Config object at 0x7f81a4ee8d10>

    @pytest.mark.slow
    def test_exceed_rate_limit(cmsetup, gencreds, maildata, chatmail_config):
        """Test that the per-account send-mail limit is exceeded."""
        user1, user2 = cmsetup.gen_users(2)
        mail = maildata(
            "encrypted.eml", from_addr=user1.addr, to_addr=user2.addr
        ).as_string()
    
        timestamps = []
        i = 0
        while len(timestamps) <= chatmail_config.max_user_send_per_minute * 1.7:
            print("Sending mail", str(i))
            i += 1
            try:
                user1.smtp.sendmail(user1.addr, [user2.addr], mail)
                timestamps.append(time.time())
            except smtplib.SMTPException as e:
                if len(timestamps) < chatmail_config.max_user_send_per_minute:
                    pytest.fail(f"rate limit was exceeded too early with msg {i}")
                outcome = e.recipients[user2.addr]
                assert outcome[0] == 450
                assert b"4.7.1: Too much mail from" in outcome[1]
                return
            timestamps[:] = [ts for ts in timestamps if ts >= (time.time() - 60)]
>       pytest.fail("Rate limit was not exceeded")
E       Failed: Rate limit was not exceeded

cmdeploy/src/cmdeploy/tests/online/test_1_basic.py:209: Failed
----------------------------- Captured stdout call -----------------------------
Creating 2 online users
smtp-connect staging2.testrun.org
smtp-login 'ac1_ah5zu@staging2.testrun.org' 'vwagmbvzu'
smtp-connect staging2.testrun.org
smtp-login 'ac2_iokeq@staging2.testrun.org' '59s8hjykf'
Sending mail 0
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 1
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 2
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 3
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 4
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 5
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 6
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 7
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 8
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 9
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 10
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 11
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 12
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 13
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 14
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 15
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 16
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 17
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 18
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 19
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 20
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 21
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 22
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 23
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 24
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 25
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 26
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 27
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 28
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 29
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 30
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 31
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 32
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 33
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 34
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 35
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 36
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 37
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 38
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 39
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 40
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 41
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 42
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 43
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 44
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 45
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 46
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 47
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 48
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 49
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 50
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 51
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 52
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 53
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 54
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 55
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 56
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 57
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 58
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 59
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 60
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 61
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 62
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 63
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 64
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 65
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 66
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 67
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 68
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 69
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 70
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 71
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 72
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 73
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 74
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 75
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 76
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 77
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 78
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 79
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 80
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 81
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 82
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 83
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 84
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 85
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 86
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 87
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 88
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 89
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 90
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 91
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 92
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 93
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 94
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 95
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 96
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 97
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 98
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 99
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 100
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 101
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
Sending mail 102
smtp-sendmail from='ac1_ah5zu@staging2.testrun.org' to_addrs=['ac2_iokeq@staging2.testrun.org']
smtp-sendmail message size: 3188
```

--- a/cmdeploy/src/cmdeploy/tests/online/test_1_basic.py
+++ b/cmdeploy/src/cmdeploy/tests/online/test_1_basic.py
@@ -190,22 +190,17 @@ def test_exceed_rate_limit(cmsetup, gencreds, maildata, chatmail_config):
         "encrypted.eml", from_addr=user1.addr, to_addr=user2.addr
     ).as_string()
 
-    timestamps = []
-    i = 0
-    while len(timestamps) <= chatmail_config.max_user_send_per_minute * 1.7:
+    for i in range(chatmail_config.max_user_send_per_minute * 3):
         print("Sending mail", str(i))
-        i += 1
         try:
             user1.smtp.sendmail(user1.addr, [user2.addr], mail)
-            timestamps.append(time.time())
         except smtplib.SMTPException as e:
-            if len(timestamps) < chatmail_config.max_user_send_per_minute:
+            if i < chatmail_config.max_user_send_per_minute:
                 pytest.fail(f"rate limit was exceeded too early with msg {i}")
             outcome = e.recipients[user2.addr]
             assert outcome[0] == 450
             assert b"4.7.1: Too much mail from" in outcome[1]
             return
-        timestamps[:] = [ts for ts in timestamps if ts >= (time.time() - 60)]
     pytest.fail("Rate limit was not exceeded")
 
 

